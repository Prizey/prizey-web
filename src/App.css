body {
  background: linear-gradient(135deg, #6243db 5%, #d9427d 50%, #de4d40 90%) no-repeat;
  background-size: cover;
}

input:-webkit-autofill {
  -webkit-transition-delay: 99999s;
}