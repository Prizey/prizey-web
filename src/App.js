import React, { Component } from 'react'
import { Provider } from 'react-redux'
import { Provider as CroodsProvider } from 'croods'
import { Auth, credentials as credentialsProps } from 'croods-auth'
import CssBaseline from '@material-ui/core/CssBaseline'
import { MuiThemeProvider } from '@material-ui/core/styles'

import LoadingComponent from 'design/Loading/Loading'
import ErrorComponent from 'design/Error/Error'

import store from 'store/store'
import FlashMessage from 'seasoned-flash'
import Router from './Router'
import theme from './theme'

import 'App.css'

export default class extends Component {
  render() {
    return (
      <FlashMessage>
        <Provider store={store}>
          <MuiThemeProvider theme={theme}>
            <CssBaseline />
            <CroodsProvider
              baseUrl={process.env.REACT_APP_API_URL}
              renderLoading={LoadingComponent}
              renderError={ErrorComponent}
              {...credentialsProps}
            >
              <Auth
                renderLoading={LoadingComponent}
                render={props => <Router {...props} />}
              />
            </CroodsProvider>
          </MuiThemeProvider>
        </Provider>
      </FlashMessage>
    )
  }
}
