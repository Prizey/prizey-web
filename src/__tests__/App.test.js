import React from 'react'
import renderer from 'react-test-renderer'
import { navigate } from '@reach/router'
import App, { checkStatus } from '../App'

jest.mock('../Router', () => () => <div>Router</div>)

jest.mock('croods', () => ({
  Info: ({ children, ...props }) => <div {...props}>Info - {children}</div>,
  List: props => <div {...props}>{props.children}</div>,
  Provider: ({ children, ...props }) => (
    <div {...props}>
      Provider - {children}
      Loading - {props.renderLoading(props)}
      Error - {props.renderError(props)}
      AfterFailure - {props.afterFailure({ status: 200 })}
    </div>
  ),
  createReducer: () => (state = {}) => state,
}))

jest.mock('@reach/router', () => ({
  navigate: jest.fn(),
}))

jest.mock('croods-auth', () => ({
  Auth: props => (
    <div {...props}>
      Auth - {props.render(props)} - {props.renderLoading()}
    </div>
  ),
  createReducer: () => (state = {}) => state,
  credentials: { foo: 'bar' },
}))

const OLD_ENV = process.env

beforeEach(() => {
  jest.resetModules()
  process.env = { ...OLD_ENV, REACT_APP_API_URL: 'https://api.foobar.com' }
  delete process.env.NODE_ENV
})

afterEach(() => {
  process.env = OLD_ENV
})

it('renders correctly', () => {
  const tree = renderer.create(<App />).toJSON()
  expect(tree).toMatchSnapshot()
})

it('test checkStatus with 503', () => {
  const response = { status: 503 }
  checkStatus(response)
  expect(navigate).toHaveBeenCalledWith(`/game-down`)
})

it('test checkStatus with 403', () => {
  const response = { status: 403 }
  checkStatus(response)
  expect(navigate).toHaveBeenCalledWith(`/blocked`)
})
