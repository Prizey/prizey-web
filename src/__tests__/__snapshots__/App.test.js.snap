// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  baseUrl="https://paynow-staging.herokuapp.com"
>
  Provider - 
  <div>
    New React App
  </div>
</div>
`;
